### ISSUE TITLE
Reassignability (`var`)

### ISSUE THREAD

**Issue by staycoolcall911** [winglang/wing#109]
https://github.com/winglang/wing/blob/main/docs/04-reference/winglang-spec.md#16-reassignability

**Comment by staycoolcall911** [comment]
Reassignability for variables was implemented in #1180 as `let var x = 5;`.
We still need to implement reassignability for class/resource members:
```ts
resource Foo {
  let bucket: cloud.Bucket;
  var name: str?;
}
```

**Comment by ekeren** [comment]
update syntax, the `var` is a modifier and the `let` is not requires for class/resource members

**Comment by ekeren** [comment]
```[tasklist]
### Tasks
- [ ] https://github.com/octo-org/octo-repo/issues/45
- [ ] Draft issue title
```

**Comment by staycoolcall911** [comment]
Closed via #1196
